motr/obj.c

```diff
@@ -555,6 +555,27 @@ static int obj_namei_op_init(struct m0_entity *entity,
 	return M0_ERR(rc);
 }
 
+/**
+ * Finds and sets optimal layout id according to the object's size.
+ */
+static void obj_optimal_lid_set(struct m0_obj *obj,
+				struct m0_layout_domain *ldom)
+{
+	uint64_t                *lid;
+
+	lid = &obj->ob_attr.oa_layout_id;
+
+	/** Find optimal layout id when pver id is set and layout id is not */
+	if (obj->ob_attr.oa_layout_id <= 0 &&
+		m0_fid_is_set(&obj->ob_attr.oa_pver))
+		*lid = m0_layout_find_by_buffsize(ldom, &obj->ob_attr.oa_pver,
+						  obj->ob_attr.oa_obj_size);
+	/** Set default layout id when both layout id and pver id is unset */
+	else if (obj->ob_attr.oa_layout_id <= 0 &&
+			!m0_fid_is_set(&obj->ob_attr.oa_pver))
+		*lid = M0_DEFAULT_LAYOUT_ID;
```

Comment:
* braces around single statement blocks can be optionally omitted;

But, in this case, this is very confusing and error-prune.
There are multiple lines of code (including comments).
So, it's better to add braces here.

Within a function, no need to have Doxygen comment.
Here, we can have,
```
/* Set default layout id when both layout id and pver id is unset */
```

Also, this comment separate the "else if" condition. Please put it in braces.

Reply:
Okay, will add those and modify the comment.

Reply:
Done.